### PR TITLE
4.1.0

### PR DESCRIPTION
**Pull request by adlarkin**
Prepare for 4.1.0 release.

The motivation for this release is to add the wheel slip plugin to Dome for https://github.com/osrf/subt/tree/dome.

Comparison to 4.0.0: https://github.com/ignitionrobotics/ign-gazebo/compare/ignition-gazebo4_4.0.0...ign-gazebo4

Signed-off-by: Ashton Larkin <ashton@openrobotics.org>

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -2,6 +2,67 @@
...
+### Ignition Gazebo 4.1.0 (2020-12-11)
+
+1. Update Dockerfiles to use focal images
+    * [pull request 388](https://github.com/ignitionrobotics/ign-gazebo/pull/388)
+
+1. Updated source build instructions for ign-gazebo4
+    * [pull request 404](https://github.com/ignitionrobotics/ign-gazebo/pull/404)
+
+1. Revert diff_drive test changes from [pull request 314](https://github.com/ignitionrobotics/ign-gazebo/pull/314)
+    * [pull request 410](https://github.com/ignitionrobotics/ign-gazebo/pull/410)
```
Comment (adlarkin) [inline Changelog.md:14]: This is referring to a commit in #410, which is actually a forward port PR (this commit was changing something for the forward port). Should this be in the changelog, or should I take it out?

File: Changelog.md [hunk 2]
```diff
@@ -2,6 +2,67 @@
...
+1. Fix flaky RecordAndPlayback test in INTEGRATION_log_system
+    * [pull request 463](https://github.com/ignitionrobotics/ign-gazebo/pull/463)
+    * [pull request 469](https://github.com/ignitionrobotics/ign-gazebo/pull/469)
+
+1. Make PeerTracker test more robust
+    * [pull request 452](https://github.com/ignitionrobotics/ign-gazebo/pull/452)
+    * [pull request 469](https://github.com/ignitionrobotics/ign-gazebo/pull/469)
+
+1. Fixes for [pull request 463](https://github.com/ignitionrobotics/ign-gazebo/pull/463)
+    * [pull request 469](https://github.com/ignitionrobotics/ign-gazebo/pull/469)
```
Comment (adlarkin) [inline Changelog.md:59]: This is referring to a commit in #469, which is actually a forward port PR (this commit was changing something for the forward port). Should this be in the changelog, or should I take it out?

**Review by adlarkin — COMMENTED** [review]
A few questions I have for any reviewers.

File: Changelog.md (hunk 1 above)
Comment (nkoenig) [reply Changelog.md:14]: You can leave this out.

File: Changelog.md (hunk 2 above)
Comment (nkoenig) [reply Changelog.md:59]: You can leave this out.

Comment (adlarkin) [reply Changelog.md]: Sounds good; removed in https://github.com/ignitionrobotics/ign-gazebo/pull/485/commits/c2237b5c6a12794e5e55fc003a000b7165da2794.

File: Changelog.md (hunk 1 above)
Comment (adlarkin) [reply Changelog.md]: Sounds good; removed in https://github.com/ignitionrobotics/ign-gazebo/pull/485/commits/c2237b5c6a12794e5e55fc003a000b7165da2794.

**Comment by adlarkin** [comment]
It looks like #229 was merged into `ign-gazebo4` shortly after making this PR. I'm waiting to update this branch to include #229 because it looks like #229 may not have been merged with the right strategy (it seems like a regular merge was done instead of a squash merge). I posted a comment in #229 asking about this and am waiting for a response: https://github.com/ignitionrobotics/ign-gazebo/pull/229#issuecomment-744520196

**Comment by nkoenig** [comment]
> It looks like #229 was merged into `ign-gazebo4` shortly after making this PR. I'm waiting to update this branch to include #229 because it looks like #229 may not have been merged with the right strategy (it seems like a regular merge was done instead of a squash merge). I posted a comment in #229 asking about this and am waiting for a response: [#229 (comment)](https://github.com/ignitionrobotics/ign-gazebo/pull/229#issuecomment-744520196)

We can fix the git history independent of a release. It's okay to move forward with the release.

**Comment by adlarkin** [comment]
> We can fix the git history independent of a release. It's okay to move forward with the release. Can you add in the changelog entry for #229?

Added #229 to the changelog in https://github.com/ignitionrobotics/ign-gazebo/pull/485/commits/1e80914b523cdfc30463e734ea040b7db030fc05. Everything should be up to date with `ign-gazebo4` now.

**Review by nkoenig — APPROVED** [review]

**Review by chapulina — APPROVED** [review]
Thanks, @adlarkin ! 

> #229 

We should try to always squash-merge non-porting PRs, but when those slip in without squashing, we shouldn't go back and rewrite history in a release branch (in this case, `ign-gazebo4`), because that could be disruptive to everyone's workflows and open PRs. So we can proceed as it is.

---

Before triggering the release, please be sure to open a PR against https://github.com/ignition-release/ign-gazebo4-release bumping the minimum required versions for dependencies. If you look in the comparison link you posted, we need to bump `ign-physics` to 3.1.0 and `ign-renderong` to 4.1.0.

**Comment by adlarkin** [comment]
> Before triggering the release, please be sure to open a PR against https://github.com/ignition-release/ign-gazebo4-release bumping the minimum required versions for dependencies. If you look in the comparison link you posted, we need to bump ign-physics to 3.1.0 and ign-rendering to 4.1.0.

I saw [this commit](https://github.com/ignitionrobotics/ign-gazebo/pull/459/commits/4e5f1d581c4d437da14fb138f99defff073a9503) that indicates the dependency on `ign-rendering4.1.0`, but can't seem to find the `ign-physics3.1.0` dependency. How did you find that dependency update? (I want to make sure I'm aware for future releases)

**Comment by adlarkin** [comment]
It looks like a few checks failed (GitHub actions and Jenkins Ubuntu CI) - I'm double checking the failures to make sure nothing is problematic for this release.

**Comment by adlarkin** [comment]
> It looks like a few checks failed (GitHub actions and Jenkins Ubuntu CI) - I'm double checking the failures to make sure nothing is problematic for this release.

There were two tests that failed, and the failures for GitHub actions and Jenkins appear to be the same. The failures both seem to be related to fuel model paths, and it looks like these failures were already occurring for several builds in Jenkins. Here are the Jenkins failures: https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/4766/testReport/

This seems like an odd failure to me, and the fact that fuel model tests are failing may be problematic for SubT. Should we try to resolve these test failures before moving forward with this release?

**Comment by nkoenig** [comment]
> This seems like an odd failure to me, and the fact that fuel model tests are failing may be problematic for SubT. Should we try to resolve these test failures before moving forward with this release?

No, we can resolve test failures in a separate PR. Please go ahead with the release.

**Comment by chapulina** [comment]
>  The failures both seem to be related to fuel model paths

They should be fixed when https://github.com/ignitionrobotics/ign-gazebo/pull/480 is forward-ported.

**Pull request merged**